src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -285,6 +289,77 @@ public void execute()
         }
     };
 
+    public static final StartupCheck checkReadAheadKbSetting = new StartupCheck()
+    {
+        // This value is in KB.
+        private static final long MAX_RECOMMENDED_READ_AHEAD_KB_SETTING = 128;
+
+        @Override
+        public void execute() throws StartupException
+        {
+            if (!FBUtilities.isLinux)
+                return;
+
+            String[] dataDirectories = DatabaseDescriptor.getRawConfig().data_file_directories;
```

Comment:
To improve readability and testability, maybe extract this to a different method?
Something like:
```java
Map<String, String> blockDevices = getBlockDevices(DatabaseDescriptor.getRawConfig().data_file_directories);
```

Also perhaps add a simple test with a temporary directory (see: https://www.javacodegeeks.com/2021/03/creating-temporary-files-with-junit-5.html).

Reply:
@pauloricardomg can you please expand on this 'Also perhaps add a simple test with a temporary directory' , is the test for the getBlockDevices which will just return the Map?

Reply:
the idea is to verify that running `getBlockDevices(new String[]{"/tmp/mytmpdir"})` will correctly return the block device associated with this temporary directory.